File: javascript/hamming/hamming.js
export const compute = (a, b) => {
    if (a.length != b.length) {
        if (!a.length)
            throw "left strand must not be empty";
        else if (!b.length)
            throw "right strand must not be empty";
        else
            throw "left and right strands must be of equal length";
    }

    let distance = 0;

    for (let i = 0; i < a.length; ++i)
        if (a[i] != b[i])
            distance += 1;

    return distance;
};

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    private int distance = 0;

    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.isEmpty() && rightStrand.isEmpty())
            return;

        if (leftStrand.isEmpty())
            throw new IllegalArgumentException("left strand must not be empty.");

        if (rightStrand.isEmpty())
            throw new IllegalArgumentException("right strand must not be empty.");

        if (leftStrand.length() != rightStrand.length())
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");

        for (var i = 0; i < leftStrand.length(); ++i) {
            if (leftStrand.charAt(i) != rightStrand.charAt(i))
                distance += 1;
        }
    }

    public int getHammingDistance() {
        return distance;
    }
}
